src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3899,4 +3807,169 @@ bool ViewProviderSketch::isInEditMode() const
 {
     return editCoinManager != nullptr;
 }
+void ViewProviderSketch::generateContextMenu()
```

Comment:
I'm OK with merging this code as-is, and I know there is a LONG history in FreeCAD of methods that look like this, but IMO it should really be something like eight individual methods, so would look something like:
```
void ViewProviderSketch::generateContextMenu()
{
    // I'm using structured bindings here, but you could write countSelection in a number of different ways
    // to achieve the same basic idea...
    const auto [selectedEdges, selectedLines, selectedConics, selectedPoints, selectedConstraints] = countSelection();
    if (selectedEdges >= 1 && selectedPoints == 0) {
        setContextMenuForEdges(menu, selectedConics, selectedLines);
    } else if (selectedEdges == 1 && selectedPoints >= 1) {
        setContextMenuForOneEdgeAndPoints(menu, selectedConics, selectedPoints);
    } 
    // and so forth, for each of your top-level conditionals -- the functions can be local 
    // to the CPP file, in an anonymous namespace, since they don't need access to `this`
}
```

Reply:
@chennes shall I redo it or do it like this for future PRs?

Reply:
But then you are creating a lot of functions which are cluttering VPSketch. Even if only the cpp file.
So the solution might be to create a class in another file to handle this. But it seems a bit overkill. And also Max is a new c++ contributor asking him that wouldn't be easy.
So perhaps we can keep like this for now? This function is long but only because of the menu names taking many lines not because of complexity.

Reply:
I'd also say that this is fine. Every branch of the if consists basically of one statement - which is pretty long but it still is one statement. Introducing more functions into that case would be harder to maintain as every added or removed scenario will require adding or removing a function which won't be reusable anyways as it is so specific.